server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -1176,16 +1245,25 @@ public NumericType numericType() {
             return type.numericType();
         }
 
+        @Override
+        public boolean isSearchable() {
+            return isIndexed() || hasDocValues();
+        }
+
         @Override
         public Query termQuery(Object value, SearchExecutionContext context) {
-            failIfNotIndexed();
-            return type.termQuery(name(), value);
+            failIfNotIndexedNorDocValues();
```

Comment:
Do you think we should have an expensive query check here as well?

Reply:
Works for me. Added check in bae5818